fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -280,13 +277,9 @@ public Response deleteObject() {
 
         try {
             resource().delete();
-
-            try {
-                session.save();
-            } catch (final RepositoryException e) {
-                throw new RepositoryRuntimeException(e);
+            if (!batchService.exists(session.getId(), getUserPrincipal())) {
```

Comment:
@awoods I have been thinking about this, and I think you make an excellent point: that much of this could be handled via runtime polymorphism. This would eliminate the awkward `if` construct, and make the implementation significantly cleaner. This will require some changes to the underlying `FedoraSession` interface, but I will add that change in a forthcoming commit.